### DISCUSSION TITLE
如何合并一个操作产生的多条历史记录

### DISCUSSION THREAD

**Discussion by xhlichen163** [antvis/X6#951] [Q&A] [answered]
在解除组合元素时，会记录多条历史队列，导致undo时需要点击很多步，如何合并？
// 只选中组时，不删除子元素
      const groups = cells.filter((item) => item.prop("type") === "group");
      groups.forEach((group) => {
        group.children.forEach((child) => {
          group.unembed(child, true);
        });
      });
      this.graph.removeCells(cells);

// 历史记录面板
       graph.history.on('change', (args) => { 
          // code here
          console.log('历史记录变化', args)
        })

**Comment by NewByVector** [comment, marked as answer]
可以将批量操作放在一个事务中，这样只记录一次 history：
```ts
graph.startBatch('name')
// do something
graph.stopBatch('name')
```
